### ISSUE TITLE
Improving label accuracy — use with different models?

### ISSUE THREAD

**Issue by stevengoldberg** [gev2002/react-native-vision-camera-image-labeler#1]
Thanks for this plugin — I implemented it and it's working very smoothly! However, in my experiments so far, the quality of the results it returns is hit or miss (it labels a lot of things "Curtain", for example). Would it be possible / easy to experiment with other models besides MLKit? Or to let the consumer specify their own model?

**Comment by Jahagirds** [comment]
HI @stevengoldberg  Are you able to compile it in iOS?

**Comment by stevengoldberg** [comment]
> HI @stevengoldberg Are you able to compile it in iOS?

@Jahagirds not exactly sure what you mean — I have successfully installed this library and built my app on iOS. I'm looking for ways to get more accurate results and wondering if the issue is with the model being used.

**Comment by Jahagirds** [comment]
@stevengoldberg I was just asking is it working fine or not, I'm still getting errors which I mentioned in issue #2 and also I'm working on simulators, 

Is it working for simulator?

**Comment by stevengoldberg** [comment]
Another shot in the dark — could something be wrong with the [getOrientation](https://github.com/gev2002/react-native-vision-camera-image-labeler/blob/main/ios/VisionCameraImageLabeler.swift#L47) function? I'm out of my depth with Swift, but `frame.orientation` logs "portrait" in my frameProcessor function, and [the example in the MLKit docs](https://developers.google.com/ml-kit/vision/image-labeling/ios#1.-prepare-the-input-image) looks a little different.

@Jahagirds I commented on your issue, let's keep them separate?

**Comment by gev2002** [comment]
Yes, you are right I modified becouse  Vision Camera not fully Supporting Orientation https://github.com/mrousavy/react-native-vision-camera/issues/1891. When supports all I will update that.

**Comment by stevengoldberg** [comment]
I see — to be clear I have no idea if orientation is a problem, I’m just looking for ways to get more accurate/useful labels. 

My app only ever takes pictures in portrait mode with the front or back camera. Do you think the orientation is reducing the quality of the labeling? If that’s not it, are there any other options to experiment with? What about different ML models?

**Comment by stevengoldberg** [comment]
Update — I discovered that the issue I was encountering only happened with some camera formats. On those, the labeler produced garbage output; almost always "Curtain, Pattern, Textile," regardless of the input. With other formats, it produced more reasonable output.

However, once I passed `pixelFormat="rbg"` to the Camera component, both formats started producing the same results (it was "yuv" before).

So that seems to solve my immediate problem. It would still be neat to try dropping in different ML models, however!
